### PR TITLE
Cxi loader

### PR DESCRIPTION
**Pull request by JulReinhardt**
Basic functionality, even for multiple entries for cxi files, however some things need to be fixed:

- [ ] lazy loading 
- [x] allow different units
- [x] define positioner axis names
- [ ] add (optional) processed data
- [ ] add other details

### CODE REVIEW

**Comment by JulReinhardt** [comment]
I've fleshed it out a bit. Maybe you can review @carterbox and @prjemian and we merge this example and Daniel's into master to go from there?

**Review by carterbox — COMMENTED** [review]
There's a lot going on here. Could you please clarify what the main objective of this PR is?

**Comment by JulReinhardt** [comment]
Yes, I know. I started mixing as I thought creating too many branches doesn't make sense either. It's basically to add more functionality to the creator class and have it used in the cxi_converter as an example case.

**Comment by carterbox** [comment]
Rebased this branch onto `main` so the git history is less confusing.

File: creator/nx_creator_ptycho.py [hunk 1]
```diff
@@ -153,113 +198,214 @@ def create_entry_group(self,
...
         return instrument_group
 
     def create_beam_group(self,
-                          incident_energy: float,
-                          entry_number: int = 1,
+                          h5parent: h5py.Group,
+                          incident_beam_energy: float,
+                          energy_units: str,
+                          wavelength_units: str = None,
+                          extent_untis: str = None,
```
Comment (carterbox) [inline creator/nx_creator_ptycho.py:234]:
```suggestion
                          extent_units: str = None,
```

File: creator/cxi_converter.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+                                units="m",
+                                depends_on="x_translation")
+            creator.create_axis(transformation=transformation,
+                                axis_name='z_translation',
+                                value=34,
+                                transformation_type='translation',
+                                vector=np.array([0, 0, 1], dtype=float),
+                                offset=np.zeros(3, dtype=float),
+                                units="m",
+                                depends_on="y_translation")
```
Comment (carterbox) [inline creator/cxi_converter.py:129]: I'm assuming these transformations are placeholders because none of them make sense.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: creator/cxi_converter.py (hunk 2 above)
Comment (JulReinhardt) [reply creator/cxi_converter.py:129]: That is correct, I do not have actual values for these axes. So I removed the value unless they actually contain any real data such as for the fine sample translations, where I link to the positioner group.

**Comment by JulReinhardt** [comment]
@carterbox I've pushed the changes you requested in a new PR on the cxi_loader branch: https://github.com/lightsources/NX-cxi-ptycho/pull/17

**Pull request merged**